lib/rules/no-multiple-template-root.js

```diff
@@ -6,6 +6,58 @@
 
 const utils = require('../utils')
 
+/**
+ * Get all comments that need to be reported
+ * @param {(HTMLComment | HTMLBogusComment | Comment)[]} comments
+ * @param {VRootElement} element
+ * @returns {(HTMLComment | HTMLBogusComment | Comment)[]}
+ */
+function getReportComments(comments, element) {
+  const commentRanges = comments.map((comment) => comment.range)
+  const elementRanges = element.children
+    .filter((child) => child.type === 'VElement')
+    .map((child) => child.range)
+
+  // should return comment directly when no any elements
+  if (elementRanges.length === 0) {
+    return comments
+  }
+
+  let commentIndex = 0
+  let elementIndex = 0
+
+  const needReportComments = []
+  while (commentIndex < commentRanges.length) {
+    const [commentStart, commentEnd] = commentRanges[commentIndex]
+    const [elementStart, elementEnd] = elementRanges[elementIndex]
+    // if the comment is in the range of element, should skip
+    if (commentStart > elementStart && commentEnd < elementEnd) {
+      commentIndex += 1
+      continue
+    }
+
+    if (commentEnd < elementStart) {
+      needReportComments.push(comments[commentIndex])
+      commentIndex += 1
+    }
+
+    // the element array has no any element, but comment still has some elements
+    if (
+      elementIndex === elementRanges.length - 1 &&
+      commentStart > elementEnd
+    ) {
+      needReportComments.push(comments[commentIndex])
+      commentIndex += 1
+    }
+
+    if (elementIndex < elementRanges.length - 1 && commentStart > elementEnd) {
+      elementIndex += 1
+    }
+  }
+
+  return needReportComments
```

Comment:
I think we can write it like this:

```suggestion
  return comments.filter(
    (comment) =>
      !elementRanges.some(
        (range) => range[0] <= comment.range[0] && comment.range[1] <= range[1]
      )
  )
```